README.rst

```diff
@@ -560,6 +561,83 @@ Example:
         assert cucumbers["start"] - cucumbers["eat"] == left
 
 
+Using the datatable Fixture
+----------------------------
+
+The ``datatable`` fixture allows you to utilise data tables defined in your Gherkin scenarios
+directly within your test functions. This is particularly useful for scenarios that require tabular data as input,
+enabling you to manage and manipulate this data conveniently.
+
+The ``datatable`` fixture in pytest-bdd allows you to access the data tables defined in your Gherkin scenarios.
+When you use the ``datatable`` fixture in a step definition, it will return the table as a list of lists,
+where each inner list represents a row from the table.
+
+For example, the Gherkin table:
+
+.. code-block:: gherkin
+
+    | name  | email            |
+    | John  | john@example.com |
+
+Will be returned by the ``datatable`` fixture as:
+
+.. code-block:: python
+
+    [
+        ["name", "email"],
+        ["John", "john@example.com"]
+    ]
+
+.. NOTE:: When using the datatable fixture, it is essential to ensure that the step to which it is applied
+          actually has an associated data table. If the step does not have an associated data table,
+          attempting to use the datatable fixture will raise an error.
+          Make sure that your Gherkin steps correctly reference the data table when defined.
+
+Full example:
+
+.. code-block:: gherkin
+
+    Feature: User roles and permissions
+
+      Scenario: Assigning roles to a user
+        Given the following user details:
+          | name  | email            |
+          | John  | john@example.com |
+
+        When the user is assigned the following roles:
+          | role  |
+          | Admin |
+          | Editor |
+
+        Then the user should have the following permissions:
+          | permission    | allowed |
+          | view content  | true    |
+          | edit content  | true    |
+          | delete content| false   |
+
+.. code-block:: python
+
+    from pytest_bdd import given, when, then
+
+    @given("the following user details:")
+    def _(datatable):
+        assert datatable == [["name", "email"], ["John", "john@example.com"]]
```

Comment:
Let's try to actually return something from this Given step, so that people won't get confused on why we are asserting in a given step